storage/google/cloud/storage/batch.py

```diff
@@ -314,10 +319,15 @@ def _unpack_batch_response(response, content):
         _, status, _ = status_line.split(' ', 2)
         sub_message = parser.parsestr(rest)
         payload = sub_message._payload
-        ctype = sub_message['Content-Type']
         msg_headers = dict(sub_message._headers)
-        msg_headers['status'] = status
-        headers = httplib2.Response(msg_headers)
-        if ctype and ctype.startswith('application/json'):
-            payload = json.loads(payload)
-        yield headers, payload
+        content_id = msg_headers.get('Content-ID')
+
+        subresponse = requests.Response()
+        subresponse.request = requests.Request(
+            method='BATCH',
+            url='contentid://{}'.format(content_id)).prepare()
```

Comment:
1. `exceptions.from_http_response` uses `response.request` to populate `message`, e.g. `{request.method} {request.url}: {message}`.
2. Therefore, we must specify a `request` for the `response` here. However, batch doesn't track the original request that the response belongs to.
3. Therefore, we do not know the original `method` or `url` so we can not re-construct the request. We can, however, gleam some information from the multipart response: the content-id for the batch request.
4. Therefore we generate a request using `BATCH` as the method and `content-id` as the URL.

This isn't perfect or great or ideal, but it provides more information than was previously available without needing to overhaul all of this.